### PR TITLE
Create python-build-workflow.yml

### PR DESCRIPTION
**Pull request by mahvidulAws**
*Issue #, if available:*

*Description of changes:*
This commit will create a github actions workflow to build, lint, test, and run security checks on new commits made to this package

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/python-build-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install poetry
+          make install
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: lint fix
```
Comment (rforsyth) [inline .github/workflows/python-build-workflow.yml:25]: The Makefile is set up to do all of the CI steps in one command: `make ci`

Comment (mahvidulAws) [reply .github/workflows/python-build-workflow.yml:25]: make ci runs everything right now. My thinking was that by separating out the different stages, we can allow developers to better know which component of the build process is failing

Comment (rforsyth) [reply .github/workflows/python-build-workflow.yml:25]: Yeah, not a big deal.  Maybe it depends on what the build output looks like between the two options.

In general it's better to put logic like what it means for a build to "pass" in one place and just point to it from other places that run the build.  That way future engineers don't need to grok both the Makefile and the build workflow config.  Easier debugging is a valid reason to break it up though.

Looks like you went ahead and made the change so seems like we can move forward.

**Review by rforsyth — DISMISSED** [review]
Thanks for updating this.  Let's squash the commits before merging so that the CR only results in one commit.

File: omics/transfer/download.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import socket
 from typing import IO, Any, List, Union
 
-from botocore.exceptions import IncompleteReadError, ReadTimeoutError
+from botocore.exceptions import IncompleteReadError, ReadTimeoutError  # type: ignore
```
Comment (rforsyth) [inline omics/transfer/download.py:6]: You should be able to get type annotations for these exceptions from [botocore-stubs](https://pypi.org/project/botocore-stubs/).

Just run: `poetry add botocore-stubs` and you should be able to remove the type ignore.

**Review by rforsyth — CHANGES_REQUESTED** [review]

File: .github/workflows/python-build-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        run: | 
+         make lint-fix
```
Comment (rforsyth) [inline .github/workflows/python-build-workflow.yml:27]: `lint-fix` is intended to be run by developers prior to check-in to fix any automatically fixable issues, not as a build step.

File: omics/transfer/download.py (hunk 2 above)
Comment (mahvidulAws) [reply omics/transfer/download.py]: Fixed

**Review by rforsyth — APPROVED** [review]

**Pull request merged**